diff -uprN apache-flume-1.5.2-src_vanilla/bin/flume-ng apache-flume-1.5.2-src/bin/flume-ng
--- apache-flume-1.5.2-src_vanilla/bin/flume-ng	2014-11-12 21:13:02.000000000 +0100
+++ apache-flume-1.5.2-src/bin/flume-ng	2015-03-31 16:45:01.876198626 +0200
@@ -233,10 +233,10 @@ run_flume() {
 ################################
 
 # set default params
-FLUME_CLASSPATH=""
-FLUME_JAVA_LIBRARY_PATH=""
-JAVA_OPTS="-Xmx20m"
-LD_LIBRARY_PATH=""
+#FLUME_CLASSPATH=""
+#FLUME_JAVA_LIBRARY_PATH=""
+#JAVA_OPTS="-Xmx20m"
+#LD_LIBRARY_PATH=""
 
 opt_conf=""
 opt_classpath=""
