--- jmxtrans.orig/jmxtrans.sh	2015-03-16 15:52:20.000000000 +0100
+++ jmxtrans/jmxtrans.sh	2015-03-17 17:48:55.000000000 +0100
@@ -3,23 +3,25 @@
 FILENAME=$2
 
 # If CONF_FILE not defined but /etc/sysconfig/jmxtrans found, use it (servicectl/initd)
-if [ -z "$CONF_FILE" ]; then
-    if [ -f /etc/sysconfig/jmxtrans ]; then
-       CONF_FILE=/etc/sysconfig/jmxtrans
-    fi
-fi
+#if [ -z "$CONF_FILE" ]; then
+#    if [ -f /etc/jmxtrans/conf ]; then
+#       CONF_FILE=/etc/jmxtrans/config.dist/
+#    fi
+#fi
 
 # Specify the commonly used configuration options below in a config file.
-CONF_FILE=${CONF_FILE:-"jmxtrans.conf"}
-if [ -e "$CONF_FILE" ]; then
-    . "$CONF_FILE"
-fi
+#CONF_FILE=${CONF_FILE:-"jmxtrans.config"}
+#if [ -e "$CONF_FILE" ]; then
+#    . "$CONF_FILE"
+#fi
+
+. /etc/jmxtrans/config.dist/jmxtrans.config
 
 JAVA_HOME=${JAVA_HOME:-"/usr"}
 LOG_DIR=${LOG_DIR:-"."}
 LOG_FILE=${LOG_FILE:-"/dev/null"}
 
-JAR_FILE=${JAR_FILE:-"jmxtrans-all.jar"}
+JAR_FILE=${JAR_FILE:-"/usr/lib/jmxtrans/jmxtrans-1.0.0-all.jar"}
 JSON_DIR=${JSON_DIR:-"."}
 SECONDS_BETWEEN_RUNS=${SECONDS_BETWEEN_RUNS:-"60"}
 HARDKILL_THRESHOLD=${HARDKILL_THRESHOLD:-60}
