diff -uprN kafka-0.8.1.1-src_vanilla/bin/kafka-run-class.sh kafka-0.8.1.1-src/bin/kafka-run-class.sh
--- kafka-0.8.1.1-src_vanilla/bin/kafka-run-class.sh	2014-04-22 20:37:23.000000000 +0200
+++ kafka-0.8.1.1-src/bin/kafka-run-class.sh	2015-02-18 10:48:22.658730984 +0100
@@ -23,7 +23,7 @@ fi
 base_dir=$(dirname $0)/..
 
 # create logs directory
-LOG_DIR=$base_dir/logs
+LOG_DIR=/var/log/kafka
 if [ ! -d $LOG_DIR ]; then
 	mkdir $LOG_DIR
 fi
